org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -343,9 +343,10 @@ public static void updateBinaries(IJavaProject javaProject, Map<String, IPath> l
 
 	public static Set<Path> collectBinaries(IPath projectDir, Set<String> include, Set<String> exclude, IProgressMonitor monitor) throws CoreException {
 		Set<Path> binaries = new LinkedHashSet<>();
-		Map<Path, Set<String>> includeByPrefix = groupGlobsByPrefix(projectDir, include);
-		Stream<Path> excludeResolved = exclude.stream().map(glob -> resolveGlobPath(projectDir, glob).toFile().toPath());
-		for (Path base: includeByPrefix.keySet()) {
+		Map<IPath, Set<String>> includeByPrefix = groupGlobsByPrefix(projectDir, include);
+		Stream<IPath> excludeResolved = exclude.stream().map(glob -> resolveGlobPath(projectDir, glob));
```

Comment:
In Windows, `IPath.toFile().toPath()` for glob pattern is no longer appropriate because Windows does not allow '*' in file name and the construction of `java.nio.file.Path` will perform a name check. So a glob can only be represented using either `String` or `IPath`.